services/vision/vision.go

```diff
@@ -118,10 +119,25 @@ func FindFirstName(r robot.Robot) string {
 	return ""
 }
 
-// FirstFromRobot returns the first vision service in this robot.
+// FirstFromRobot returns the first vision service on this robot.
 func FirstFromRobot(r robot.Robot) (Service, error) {
-	name := FindFirstName(r)
-	return FromRobot(r, name)
+	vis, err := FirstFromLocalRobot(r)
+	if err != nil {
+		name := FindFirstName(r)
+		return FromRobot(r, name)
+	}
+	return vis, err
+}
+
+// FirstFromLocalRobot returns the first vision service on this main robot.
+// This will specifically ignore remote resources.
+func FirstFromLocalRobot(r robot.Robot) (Service, error) {
+	for _, n := range r.ResourceNames() {
+		if n.Subtype == Subtype && !n.ContainsRemoteNames() {
+			return FromRobot(r, n.ShortName())
```

Comment:
Wasn't the error that the ShortName could resolve to the remote and load that instead? @cheukt

Reply:
since the code does make sure that the name is not of a remote resource, we should be good here. but yes, short name can resolve to remote resources some times

Reply:
got it- also talked with Khari and he explained it as well.